File: packages/zodiac-js/src/models/zodiac-sign.ts
/**
 * Western Zodiac sign model.
 *
 * Represents a single Western Zodiac sign with all associated metadata.
 */
export interface ZodiacSign {
  /** Stable, language-independent identifier (e.g., "aries") */
  readonly id: string;

  /** English display name */
  readonly name: string;

  /** Symbol name (e.g., "Ram") */
  readonly symbol: string;

  /** Unicode zodiac symbol (e.g., "♈") */
  readonly unicodeSymbol: string;

  /** Date range for this sign */
  readonly dateRange: {
    readonly startMonth: number;
    readonly startDay: number;
    readonly endMonth: number;
    readonly endDay: number;
  };

  /** Associated element ID (e.g., "fire") */
  readonly element: string;

  /** Associated modality ID (e.g., "cardinal") */
  readonly modality: string;

  /** Ruling planet information */
  readonly rulingPlanet: RulingPlanetRef;

  /** Traditional astrological description */
  readonly description: string;

  /** Characteristic keywords */
  readonly keywords: readonly string[];

  /** Traditional strengths */
  readonly strengths: readonly string[];

  /** Traditional weaknesses */
  readonly weaknesses: readonly string[];

  /** Icon identifier */
  readonly iconId: string;
}

/** Reference to ruling planet(s) with traditional/modern distinction */
export interface RulingPlanetRef {
  readonly traditional: string;
  readonly modern: string;
}

File: packages/zodiac-dart/lib/src/models/zodiac_sign.dart
/// Western Zodiac sign model.
///
/// Represents a single Western Zodiac sign with all associated metadata.
class ZodiacSign {
  /// Stable, language-independent identifier (e.g., "aries")
  final String id;

  /// English display name
  final String name;

  /// Symbol name (e.g., "Ram")
  final String symbol;

  /// Unicode zodiac symbol (e.g., "♈")
  final String unicodeSymbol;

  /// Date range for this sign
  final DateRange dateRange;

  /// Associated element ID (e.g., "fire")
  final String element;

  /// Associated modality ID (e.g., "cardinal")
  final String modality;

  /// Ruling planet information
  final RulingPlanetRef rulingPlanet;

  /// Traditional astrological description
  final String description;

  /// Characteristic keywords
  final List<String> keywords;

  /// Traditional strengths
  final List<String> strengths;

  /// Traditional weaknesses
  final List<String> weaknesses;

  /// Icon identifier
  final String iconId;

  const ZodiacSign({
    required this.id,
    required this.name,
    required this.symbol,
    required this.unicodeSymbol,
    required this.dateRange,
    required this.element,
    required this.modality,
    required this.rulingPlanet,
    required this.description,
    required this.keywords,
    required this.strengths,
    required this.weaknesses,
    required this.iconId,
  });
}

class DateRange {
  final int startMonth;
  final int startDay;
  final int endMonth;
  final int endDay;

  const DateRange({
    required this.startMonth,
    required this.startDay,
    required this.endMonth,
    required this.endDay,
  });
}

class RulingPlanetRef {
  final String traditional;
  final String modern;

  const RulingPlanetRef({
    required this.traditional,
    required this.modern,
  });
}
